### DISCUSSION TITLE
If throttling happens, rendering won't act correctly (speed for entities).

### DISCUSSION THREAD

**Discussion by Norlock** [bevyengine/bevy#4233] [Q&A] [answered]
## Bevy version
0.6.1

https://github.com/norlock/sparticles-bevy/tree/bug-more-particles
8c6bf7559b3ec8fc1a7ad3a378ef62427408b7ca

## Operating system & version
I use Arch on Wayland.

## What you did

In main.rs  there is a field I changed: 
        particles_per_emission: 200,

If you change from 20 particles_per_emission to 200 you can see it will start to drastically increase in speed.
I use delta in transform, and turned off all animations / forces to turn off external factors.

## What you expected to happen
That it won't drastically speed up if it starts to throttle but stay in range of its actual speed. I can understand if it stutters but not to speed up.

## What actually happened
It will start to speed up.

## Additional information
-

**Comment by alice-i-cecile** [comment]
> 8c6bf7559b3ec8fc1a7ad3a378ef62427408b7ca (bug-more-particles branch)

I think you intended to link to one of your repos here? I'm having a hard time understanding what you're trying to report.

**Comment by DJMcNab** [comment]
8c6bf7559b3ec8fc1a7ad3a378ef62427408b7ca isn't an object in my local clone of bevy's repository, and neither GitHub nor Google give any results for commits with that hash.
Additionally, the string `particles_per_emission` never appears in the `bevy` respository, and `"bug-more-particles" bevy` also gives no results.

**Comment by Norlock** [comment]
I added my repository I forgot, but what I'm trying to explain is that somewhere if it starts to throttle (by changing particles to 200), the movement speeds up. Maybe I'm too inexperienced with gamedev and not really grasping what is causing it. I thought the delta I added in the transform system would mitigate the changes in frame performance and that it shouldn't visually appear to speed up (in this case movement of particles).

**Comment by alice-i-cecile** [comment]
So, questions for help with specific problems belong in Discussions. I'm going to convert this to a discussion. If you can identify a clear bug in Bevy's code with a minimal reproduction, please feel free to file a new issue.

**Comment by DJMcNab** [comment, marked as answer]
What's happening in your program is the interaction between `Time` and `FixedTimeStep`.

The `delta_seconds` in `Time` is only updated every frame. 

However, due to your fixed timestep, the system `transform_particle_system` runs multiple times each frame, but with the delta between each real frame each time.

The easiest way to fix this will be to use the delta of `1./60.` instead of `time.delta_seconds()` in `transform_particle_system`.

In the future, we'll want to make this issue more difficult to run into. Basically, `Time::delta_seconds` is currently reasonably meaningless in a system controlled by a `FixedTimeStep`
